### PR TITLE
Bugfix: Expansion of escape sequences had wrong priority

### PR DESCRIPTION
**Pull request by Hrubon**
Double bask-slashes should expand first, then the quotes. See the added example in tests.

### CODE REVIEW

File: env_test.go [hunk 1]
```diff
@@ -182,6 +182,7 @@ func TestSlice(t *testing.T) {
 			`aa`,
 		},
 		`\?`: {`?`},
+		`"foo\\"`: {`foo\`},
```
Comment (graywolf-at-work) [inline env_test.go:185]: Consider also adding test line

```
`foo \\\\ bar`: {`foo \\ bar`}
```

and see what breaks :)

**Review by graywolf-at-work — CHANGES_REQUESTED** [review]
I do not think this is correct.

**Comment by Hrubon** [comment]
I refactored the slice parsing logic a little bit. However, I kept the two passes over the input (tokenization and expansion of escape sequences). I tried to implement it in one pass but it sucked in readability.

**Review by graywolf-at-work — CHANGES_REQUESTED** [review]
Looks good thx :)

Just few readability changes requested.

**Pull request merged**